target/cheri-common/cheri-translate-utils.h

```diff
@@ -258,20 +264,19 @@ static inline void gen_check_branch_target_dynamic(DisasContext *ctx, TCGv addr)
     //        call    raise_bounds_violation
     // Note: JR/JALR will often be used in hybrid/non-CHERI cases, so we can
     // skip the less than check if pcc.base is zero and top is MAX:
-    const bool need_base_check = ctx->base.pcc_base > 0;
-    const bool need_top_check = ctx->base.pcc_top != TYPE_MAXIMUM(target_ulong);
-    if (likely(!need_base_check && !need_top_check)) {
-        return;
+    if (likely(have_cheri_tb_flags(ctx, TB_FLAG_PCC_FULL_AS))) {
```

Comment:
Is that really `likely`?  I guess it is for non-CHERI code?

Reply:
Well it's only used for non-CHERI jr/jalr and (at least currently) that code runs with full address space PCC. Not sure if we want to try and run the CheriBSD kernel with a restricted PCC. I'm not sure what we do for userspace MIPS programs. It might not be likely there...

Reply:
Well the purecap kernel is going to have a restricted PCC. The likeliness is really a function of what you run on top of QEMU. So I really think we should drop the static hints and let the branch predictor learn the dynamic behaviour.

Reply:
Sure I'm happy to drop those hints. Probably won't make much of a difference.